### ISSUE TITLE
`<Calendar/>` does not trigger `onClose` with click outside and with `Esc` if not in focus

### ISSUE THREAD

**Issue by tyv** [wix/wix-style-react#5758]
# 🐛 Bug Report

### 🏗 Relevant Components

Calendar

### 😯 Current Behavior

1. Click outside of Calendar does not trigger `onClose` handler. 
I wasn't able to find code that should do that. I think it is just forgotten. 
2. Pressing `Esc` triggers `onClose` only when calendar is focused. 

### 🤔 Expected Behavior

as docs says about `onClose`: 
> Callback function called whenever user press escape or **click outside of the element** ...

### 👣 Steps to Reproduce

I tried to launch it on your codesanbox, or fork, and faced errors.
So you use it like that 

```css
.calendar {
  position: absolute;
  z-index: 1;

  padding: 10px;
  margin-top: -1px;

  background: #fff;
  border: 1px solid #C1E4FE;
  border-radius: 6px;
}
```

```tsx
<Calendar
    className={s.calendar}
    showMonthDropdown
    showYearDropdown
    shouldCloseOnSelect
    selectionMode="day"
    autoFocus={false}
    onChange={v => console.log(v)}
    onClose={() => console.log('hide')}
/>
```

here is demo: https://www.screencast.com/t/R9MisZslQ

during demo I am making many clicks outside and press `esc`

As a workaround I probably have to wrap it in some tooltip like component. 
But one of two needs to be done: or fix docs, or add proper click outside and global event handling. 

  - **WSR version**: ` 8.31.0`
  - **Yoshi version**: `4.119.0`

### 👀 Severity

- Major

**Comment by EgoziE** [comment]
We're deprecating onClickOutside functionality in Calendar.
The `<Calendar/>` component should have only the date related functionality.
 
Use the `<DatePicker/>` component in order to have clickOutside functionality.
P.S. I checked, pressing ESC closes the date picker.

**Comment by EgoziE** [comment]
If your design does not permit using the DatePicker, please contact us through the channel with your design.
